docs/standard/architecture/securing-net-microservices-and-web-applications/using-azure-key-vault-to-protect-secrets-at-production-time.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Using Azure Key Vault to protect secrets at production time | Microsoft Docs 
+description: .NET Microservices Architecture for Containerized .NET Applications | Using Azure Key Vault to protect secrets at production time
+keywords: Docker, Microservices, ASP.NET, Container
+author: CESARDELATORRE
+ms.author: wiwagn
+ms.date: 05/26/2017
+ms.prod: .net-core
+ms.technology: dotnet-docker
+---
+# Using Azure Key Vault to protect secrets at production time
+
+Secrets stored as environment variables or stored by the Secret Manager tool are still stored locally and unencrypted on the machine. A more secure option for storing secrets is [Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/), which provides a secure, central location for storing keys and secrets.
+
+The Microsoft.Extensions.Configuration.AzureKeyVault package allows an ASP.NET Core application to read configuration information from Azure Key Vault. To start using secrets from an Azure Key Vault, you follow these steps:
+
+1.  Register your application as an Azure AD application. (Access to key vaults is managed by Azure AD.) This can be done through the Azure management portal.
+
+Alternatively, if you want your application to authenticate using a certificate instead of a password or client secret, you can use the [New-AzureRmADApplication](https://docs.microsoft.com/en-us/powershell/resourcemanager/azurerm.resources/v3.3.0/new-azurermadapplication) PowerShell cmdlet. The certificate that you register with Azure Key Vault needs only your public key. (Your application will use the private key.)
+
+1.  Give the registered application access to the key vault by creating a new service principal. You can do this using the following PowerShell commands:
```

Comment:
Fixed by using text to describe the ordered steps. Code in a numbered list wasn't formatting correctly.